MCprep_addon/world_tools.py

```diff
@@ -142,22 +236,17 @@ def detect_world_exporter(filepath: Path) -> None:
 					"# File type: Export tiles for textures to directory textures",
 					"# File type: Export individual textures to directory tex"
 				)
-				print('"{}"'.format(header))
 				if header in atlas:  # If a texture atlas is used
 					obj_header.set_atlas()
 				elif header in tiles:  # If the OBJ uses individual textures
 					obj_header.set_seperated()
-				return
+				return obj_header
 		except UnicodeDecodeError:
 			print(f"Failed to read first line of obj: {filepath}")
-			return
-		obj_header.set_jmc2obj()
-		# Since this is the default for Jmc2Obj,
-		# we'll assume this is what the OBJ is using
-		obj_header.set_seperated()
+		return obj_header
 
 
-def convert_mtl(filepath):
+def convert_mtl(filepath) -> Optional[MCprepError]:
```

Comment:
I'm thinking about this further, I feel like the bool signature we had before made more sense. It's useful to know if it actually did something or not. The old comment was a little wrong, it should have said "Returns: True if success, None if skipped".

Since we have the error class, maybe we can just return bool and this means whether it made a change or not?

```suggestion
def convert_mtl(filepath) -> Union[bool, MCprepError]:
```

Reply:
That sounds good to me

Reply:
Done, I've also pushed a slight efficiency change that makes the function exit early if the current color space is in our list of "built ins" (basically what we know we works). That way, MTLs aren't unnecessarily loaded into memory at runtime.

I've also found that we didn't have AgX in our list of built ins, so I've gone ahead and added it, alongside Khronos PBR Neutral (which is coming in 4.2)